[bug] api server panics on workflow parameter without value

### What steps did you take

<!-- A clear and concise description of what the bug is.-->
1. create a run with a parameter with no values

### What happened:

KFP api server panics with the following stacktrace:

```
I0403 13:01:11.747485       7 interceptor.go:29] /api.RunService/CreateRun handler starting
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x15e7999]

goroutine 2100 [running]:
github.com/kubeflow/pipelines/backend/src/common/util.(*Workflow).OverrideParameters(0xc0000c2e28, 0xc000ebf4a0)
	/go/src/github.com/kubeflow/pipelines/backend/src/common/util/workflow.go:54 +0x2f9
github.com/kubeflow/pipelines/backend/src/apiserver/resource.(*ResourceManager).CreateRun(0xc000326ea0, 0xc0001e8fc0, 0x0, 0x0, 0xc00034c1a0)
	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/resource/resource_manager.go:362 +0x2ca
github.com/kubeflow/pipelines/backend/src/apiserver/server.(*RunServer).CreateRun(0xc0000aa6f0, 0x1fb0360, 0xc000ebf0e0, 0xc000ebf110, 0xc0000aa6f0, 0x2d58e08, 0xc0001793b0)
	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/server/run_server.go:135 +0xa6
github.com/kubeflow/pipelines/backend/api/go_client._RunService_CreateRun_Handler.func1(0x1fb0360, 0xc000ebf0e0, 0x1b97b40, 0xc000ebf110, 0xc000641ac8, 0x1, 0x1, 0x7f9af189b008)
	/go/src/github.com/kubeflow/pipelines/backend/api/go_client/run.pb.go:1501 +0x86
main.apiServerInterceptor(0x1fb0360, 0xc000ebf0e0, 0x1b97b40, 0xc000ebf110, 0xc000edb500, 0xc000edb520, 0xc000e78b58, 0x495b68, 0x1b87600, 0xc000ebf0e0)
	/go/src/github.com/kubeflow/pipelines/backend/src/apiserver/interceptor.go:30 +0xf7
github.com/kubeflow/pipelines/backend/api/go_client._RunService_CreateRun_Handler(0x1bb5100, 0xc0000aa6f0, 0x1fb0360, 0xc000ebf0e0, 0xc0007c69c0, 0x1d96158, 0x1fb0360, 0xc000ebf0e0, 0xc001278000, 0x2ddac)
	/go/src/github.com/kubeflow/pipelines/backend/api/go_client/run.pb.go:1503 +0x14b
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00059e000, 0x1fd4720, 0xc000206c00, 0xc000416f00, 0xc00055e150, 0x2fa68a0, 0x0, 0x0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.34.0/server.go:1210 +0x4fd
google.golang.org/grpc.(*Server).handleStream(0xc00059e000, 0x1fd4720, 0xc000206c00, 0xc000416f00, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.34.0/server.go:1533 +0xd57
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000290720, 0xc00059e000, 0x1fd4720, 0xc000206c00, 0xc000416f00)
	/go/pkg/mod/google.golang.org/grpc@v1.34.0/server.go:871 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.34.0/server.go:869 +0x204
```

### What did you expect to happen:

### Environment:

<!-- Please fill in those that seem relevant. -->

*  How do you deploy Kubeflow Pipelines (KFP)? 
<!-- For more information, see an overview of KFP installation options: https://www.kubeflow.org/docs/pipelines/installation/overview/. -->
*  KFP version: 1.5.0-rc.0
<!-- Specify the version of Kubeflow Pipelines that you are using. The version number appears in the left side navigation of user interface.
To find the version number, See version number shows on bottom of KFP UI left sidenav. -->


### Anything else you would like to add:
<!-- Miscellaneous information that will assist in solving the issue.-->


### Labels
<!-- Please include labels below by uncommenting them to help us better triage issues -->

<!-- /area frontend -->
/area backend
<!-- /area sdk -->
<!-- /area testing -->
<!-- /area samples -->
<!-- /area components -->


---

<!-- Don't delete message below to encourage users to support your issue! -->
Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

--- a/backend/src/common/util/workflow.go
+++ b/backend/src/common/util/workflow.go
@@ -50,7 +50,7 @@ func (w *Workflow) OverrideParameters(desiredParams map[string]string) {
 		var desiredValue *string = nil
 		if param, ok := desiredParams[currentParam.Name]; ok {
 			desiredValue = &param
-		} else {
+		} else if currentParam.Value != nil {
 			desired := currentParam.Value.String()
 			desiredValue = &desired
 		}

--- a/backend/src/common/util/workflow_test.go
+++ b/backend/src/common/util/workflow_test.go
@@ -180,48 +180,120 @@ func TestWorkflow_OverrideName(t *testing.T) {
 }
 
 func TestWorkflow_OverrideParameters(t *testing.T) {
-	workflow := NewWorkflow(&workflowapi.Workflow{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "WORKFLOW_NAME",
-		},
-		Spec: workflowapi.WorkflowSpec{
-			Arguments: workflowapi.Arguments{
-				Parameters: []workflowapi.Parameter{
-					{Name: "PARAM1", Value: workflowapi.AnyStringPtr("VALUE1")},
-					{Name: "PARAM2", Value: workflowapi.AnyStringPtr("VALUE2")},
-					{Name: "PARAM3", Value: workflowapi.AnyStringPtr("VALUE3")},
-					{Name: "PARAM4", Value: workflowapi.AnyStringPtr("")},
-					{Name: "PARAM5", Value: workflowapi.AnyStringPtr("VALUE5")},
+	var tests = []struct {
+		name      string
+		workflow  *workflowapi.Workflow
+		overrides map[string]string
+		expected  *workflowapi.Workflow
+	}{
+		{
+			name: "override parameters",
+			workflow: &workflowapi.Workflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "WORKFLOW_NAME",
+				},
+				Spec: workflowapi.WorkflowSpec{
+					Arguments: workflowapi.Arguments{
+						Parameters: []workflowapi.Parameter{
+							{Name: "PARAM1", Value: workflowapi.AnyStringPtr("VALUE1")},
+							{Name: "PARAM2", Value: workflowapi.AnyStringPtr("VALUE2")},
+							{Name: "PARAM3", Value: workflowapi.AnyStringPtr("VALUE3")},
+							{Name: "PARAM4", Value: workflowapi.AnyStringPtr("")},
+							{Name: "PARAM5", Value: workflowapi.AnyStringPtr("VALUE5")},
+						},
+					},
+				},
+			},
+			overrides: map[string]string{
+				"PARAM1": "NEW_VALUE1",
+				"PARAM3": "NEW_VALUE3",
+				"PARAM4": "NEW_VALUE4",
+				"PARAM5": "",
+				"PARAM9": "NEW_VALUE9",
+			},
+			expected: &workflowapi.Workflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "WORKFLOW_NAME",
+				},
+				Spec: workflowapi.WorkflowSpec{
+					Arguments: workflowapi.Arguments{
+						Parameters: []workflowapi.Parameter{
+							{Name: "PARAM1", Value: workflowapi.AnyStringPtr("NEW_VALUE1")},
+							{Name: "PARAM2", Value: workflowapi.AnyStringPtr("VALUE2")},
+							{Name: "PARAM3", Value: workflowapi.AnyStringPtr("NEW_VALUE3")},
+							{Name: "PARAM4", Value: workflowapi.AnyStringPtr("NEW_VALUE4")},
+							{Name: "PARAM5", Value: workflowapi.AnyStringPtr("")},
+						},
+					},
 				},
 			},
 		},
-	})
-
-	workflow.OverrideParameters(map[string]string{
-		"PARAM1": "NEW_VALUE1",
-		"PARAM3": "NEW_VALUE3",
-		"PARAM4": "NEW_VALUE4",
-		"PARAM5": "",
-		"PARAM9": "NEW_VALUE9",
-	})
-
-	expected := &workflowapi.Workflow{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "WORKFLOW_NAME",
+		{
+			name: "handles missing parameter values",
+			workflow: &workflowapi.Workflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "NAME",
+				},
+				Spec: workflowapi.WorkflowSpec{
+					Arguments: workflowapi.Arguments{
+						Parameters: []workflowapi.Parameter{
+							{Name: "PARAM1"}, // note, there's no value here
+						},
+					},
+				},
+			},
+			overrides: nil,
+			expected: &workflowapi.Workflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "NAME",
+				},
+				Spec: workflowapi.WorkflowSpec{
+					Arguments: workflowapi.Arguments{
+						Parameters: []workflowapi.Parameter{
+							{Name: "PARAM1"},
+						},
+					},
+				},
+			},
 		},
-		Spec: workflowapi.WorkflowSpec{
-			Arguments: workflowapi.Arguments{
-				Parameters: []workflowapi.Parameter{
-					{Name: "PARAM1", Value: workflowapi.AnyStringPtr("NEW_VALUE1")},
-					{Name: "PARAM2", Value: workflowapi.AnyStringPtr("VALUE2")},
-					{Name: "PARAM3", Value: workflowapi.AnyStringPtr("NEW_VALUE3")},
-					{Name: "PARAM4", Value: workflowapi.AnyStringPtr("NEW_VALUE4")},
-					{Name: "PARAM5", Value: workflowapi.AnyStringPtr("")},
+		{
+			name: "overrides a missing parameter value",
+			workflow: &workflowapi.Workflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "NAME",
+				},
+				Spec: workflowapi.WorkflowSpec{
+					Arguments: workflowapi.Arguments{
+						Parameters: []workflowapi.Parameter{
+							{Name: "PARAM1"}, // note, there's no value here
+						},
+					},
+				},
+			},
+			overrides: map[string]string{
+				"PARAM1": "VALUE1",
+			},
+			expected: &workflowapi.Workflow{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "NAME",
+				},
+				Spec: workflowapi.WorkflowSpec{
+					Arguments: workflowapi.Arguments{
+						Parameters: []workflowapi.Parameter{
+							{Name: "PARAM1", Value: workflowapi.AnyStringPtr("VALUE1")},
+						},
+					},
 				},
 			},
 		},
 	}
-	assert.Equal(t, expected, workflow.Get())
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			workflow := NewWorkflow(tt.workflow)
+			workflow.OverrideParameters(tt.overrides)
+			assert.Equal(t, tt.expected, workflow.Get())
+		})
+	}
 }
 
 func TestWorkflow_SetOwnerReferences(t *testing.T) {

